File: 2022-2023/sourcecode/lab/lab9/RoomRelationsJava/app/src/main/java/pl/edu/uwr/pum/roomrelationsjava/model/entities/relations/LectureWithStudents.java
package pl.edu.uwr.pum.roomrelationsjava.model.entities.relations;

import androidx.room.Embedded;
import androidx.room.Junction;
import androidx.room.Relation;

import java.util.List;

import pl.edu.uwr.pum.roomrelationsjava.model.entities.Lecture;
import pl.edu.uwr.pum.roomrelationsjava.model.entities.Student;

public class LectureWithStudents {
    @Embedded
    public Lecture lecture;

    @Relation(
            parentColumn = "lectureName",
            entityColumn = "studentName",
            associateBy = @Junction(StudentLectureCrossRef.class)
    )

    public List<Student> studentList;
}

File: 2022-2023/sourcecode/lab/lab9/RoomRelationsKotlin/app/src/main/java/pl/edu/uwr/pum/roomrelationskotlin/model/entity/relations/LectureWithStudents.kt
package pl.edu.uwr.pum.roomrelationskotlin.model.entity.relations

import androidx.room.Embedded
import androidx.room.Junction
import androidx.room.Relation
import pl.edu.uwr.pum.roomrelationskotlin.model.entity.Lecture
import pl.edu.uwr.pum.roomrelationskotlin.model.entity.Student

data class LectureWithStudents (
    @Embedded val lecture: Lecture,

    @Relation(
        parentColumn = "lectureName",
        entityColumn = "studentName",
        associateBy = Junction(StudentLectureCrossRef::class)
    )

    val students: List<Student>
        )